pyiron_base/jobs/job/extension/server/queuestatus.py

```diff
@@ -295,29 +301,36 @@ def update_from_remote(project, recursive=True, ignore_exceptions=False):
         else:
             jobs_now_running_lst = []
         failed_jobs = []
-        for job_id in df_combined.id.values:
-            if job_id not in jobs_now_running_lst:
-                try:
-                    job = project.inspect(job_id)
-                    state.queue_adapter.transfer_file_to_remote(
-                        file=job.project_hdf5.file_name,
-                        transfer_back=True,
-                        delete_remote=False,
-                    )
-                    status_hdf5 = job.project_hdf5["status"]
-                    project.set_job_status(job_specifier=job.job_id, status=status_hdf5)
-                    if status_hdf5 in job_status_finished_lst:
-                        job_object = job.to_object()
-                        job_object.transfer_from_remote()
-                except Exception as e:
-                    if ignore_exceptions:
+        fetch_ids = df_combined.id.values[
+            ~np.isin(df_combined.id.values, df_queue.jobid.values)
+        ]
+        for job_id in fetch_ids:
+            try:
+                job = project.load(job_id)
+                job.transfer_from_remote()
+                if job.status in job_status_finished_lst:
+                    continue
+
+                if try_collecting:
+                    try:
+                        job.status.collect = True
+                        job.run()
+                    except Exception as e:
                         state.logger.warning(
-                            f"An error occured while trying to retrieve job {job_id}\n"
+                            f"An error occured while trying to collect job {job_id}\n"
                             f"Error message: \n{e}"
                         )
-                        failed_jobs.append(job_id)
-                    else:
-                        raise e
+                    failed_jobs.append(job_id)
```

Comment:
This is my first time reading this file, and I found that some of these chunks are quite long and one really has to sit down and read the whole thing to understand anything. IMO it would be nice to refactor some of these clauses into smaller methods. We don't need to do that for the opening of the function in this PR, but we could do it here for the changes.

e.g.
```
try:
    failure_id = self.update_remote_job(project.load(job_id), try_collecting=try_collecting)
    if failure_id is not None:
        failed_jobs.append(failure_id)
except:
    ....
```

```
def update_remote_job(job: GenericJob, try_collecting: bool = True):
    """
    Transfers a job from a remote source, and if requested, attempts to run collection
    on all unfinished jobs.
    
    Args:
        job (GenericJob): The job to update, must have been run remotely(?)
        try_collect (bool): If the job does not have a valid finished status, try setting the
            status to `collect` and running again. (Default is True.)

    Returns:
        (None | int): The job id if an exception is encountered.
    """
    job.transfer_from_remote()
    if job.status in job_status_finished_lst:
        return

    if try_collecting:
        try:
            job.status.collect = True
            job.run()
            return
        except Exception as e:
            state.logger.warning(f"An error occurred while trying to collect job {job.id}\n")
            return job.id
```